### PR TITLE
PERF: Allow user serializer to make use of preloaded custom fields

### PR DESCRIPTION
**Pull request by davidtaylorhq**
This will be used to allow serializing multiple user cards without N+1 queries

### CODE REVIEW

File: app/serializers/user_card_serializer.rb [hunk 1]
```diff
@@ -197,4 +197,11 @@ def featured_topic
   def card_background_upload_url
     object.card_background_upload&.url
   end
+
+  private
+
+  def custom_field_keys
+    # Can be extended by other serializers
```
Comment (eviltrout) [inline app/serializers/user_card_serializer.rb]: If it's meant to be extended shouldn't the method be `protected`?

Comment (eviltrout) [reply app/serializers/user_card_serializer.rb]: Also this isn't calling `super` so isn't the logic above about `can_edit` ignored?

Comment (davidtaylorhq) [reply app/serializers/user_card_serializer.rb]: I don't follow, why is `protected` required for inheritence?

Also, the inheritence goes `UserSerializer < UserCardSerializer`, so `UserSerializer` is the one which calls `super`.

Comment (eviltrout) [reply app/serializers/user_card_serializer.rb]: Thanks I assumed that card extended user and not the inverse.

As for protected vs private I think that's my Java/C++ background. I always use `protected` to indicate that a method is meant to be used by the class and all its subclasses. In ruby you're right it doesn't matter, private allows that.

`protected` is somewhat useless in ruby I think as a result :) You can keep it this way if you like and consider my labelling antiquated :)

**Pull request merged**